packages/playwright-core/src/client/elementHandle.ts

```diff
@@ -279,14 +281,26 @@ export async function convertInputFiles(files: string | FilePayload | string[] |
       }));
       return { streams };
     }
-    return { localPaths: items.map(f => path.resolve(f as string)) as string[] };
+    if (context._isLocalBrowser())
+      return { localPaths: items.map(f => path.resolve(f as string)) as string[] };
+    const filePayloads: SetInputFilesFiles = await Promise.all((items as string[]).map(async item => {
+      return {
+        name: path.basename(item),
+        buffer: await fs.promises.readFile(item)
+      };
```

Comment:
nit: ideally we pass lastModifiedMs over here to the server an use it like [here](https://github.com/microsoft/playwright/pull/27798/files#diff-06edc6b9e1c43786236ac754ec9a7ed6a4470c550a657aa6d38c2afc5dbdd850).